File: OtherAlgorithmStudy/Hackerrank/Implementation/DesignerPDFViewer.cpp
//
//  DesignerPDFViewer.cpp
//  OtherAlgorithmStudy
//
//  Created by MinKyeongTae on 26/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

/// MARK: - Designer PDF Viewer : Implementation Problem

//  Sample Input)
//  1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5
//  abc
//
//  Sample Output)
//  9

#include <stdio.h>
#include <vector>
#include <string>

using namespace std;

int designerPdfViewer(vector<int> h, string word) {
    int MAX=0;
    for(int i=0; i<word.length(); i++) MAX = MAX < h[word[i]-'a'] ? h[word[i]-'a'] : MAX;
    return MAX*int(word.length());
}

File: OtherAlgorithmStudy/Hackerrank/Implementation/DesignerPDFViewer.swift
//
//  DesignerPDFViewer.swift
//  OtherAlgorithmStudy
//
//  Created by MinKyeongTae on 26/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

// MARK: - Designer PDF Viewer : Implementation Problem

//  Sample Input)
//  1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5
//  abc
//
//  Sample Output)
//  9

import Foundation

func designerPdfViewer(h: [Int], word: String) -> Int {
    let arr = word.sorted()
    var MAX = 0
    for i in arr.indices {
        let chIdx = Int(arr[i].unicodeScalars.first!.value - Unicode.Scalar("a").value)
        MAX = MAX < h[chIdx] ? h[chIdx] : MAX
    }
    return MAX * arr.count
}
